src/main/java/uk/gov/hmcts/befta/util/MapVerifier.java

```diff
@@ -226,6 +245,39 @@ private Boolean canAcceptNullFor(Object expectedValue) {
         if (!(expectedValue instanceof String)) {
             return Boolean.FALSE;
         }
-        return !NOT_NULL.equalsIgnoreCase((String) expectedValue);
+        String strExpectedValue = (String) expectedValue;
+        return nullable(strExpectedValue) || !notNullable(strExpectedValue);
+    }
+
+    private boolean nullable(String strExpectedValue) {
+        return ANY_INTEGER_NULLABLE.getValue().equalsIgnoreCase(strExpectedValue) ||
+                ANY_DATE_NULLABLE.getValue().equalsIgnoreCase(strExpectedValue) ||
+                ANY_FLOATING_NULLABLE.getValue().equalsIgnoreCase(strExpectedValue) ||
+                ANY_NUMBER_NULLABLE.getValue().equalsIgnoreCase(strExpectedValue) ||
+                ANY_STRING_NULLABLE.getValue().equalsIgnoreCase(strExpectedValue) ||
+                ANY_OBJECT_NULLABLE.getValue().equalsIgnoreCase(strExpectedValue) ||
+                ANY_TIMESTAMP_NULLABLE.getValue().equalsIgnoreCase(strExpectedValue) ||
+                ANY_NULLABLE.getValue().equalsIgnoreCase(strExpectedValue) ||
+                ANYTHING_IF_EXISTS.getValue().equalsIgnoreCase(strExpectedValue);
+    }
```

Comment:
This and notNullable(..) method could be implemented more efficiently. We can have a boolean nullable field in the enum elements, pass a value in element constructors, and then here, we can get the element from strExpectedValue with a method like MapPlaceholder.getByValue(String value) from the enum type and check with its isNullable() method.

Reply:
Changed the implementation